### PR TITLE
Feature/3323 stylesheet for mdx

### PR DESCRIPTION
**Pull request by bkarambe**
Fixes # BLUI-3323

#### Changes proposed in this Pull Request:
- Added markdownMapping and markdown.css

#### Screenshots:
Screenshot 2022-08-02 at 4 30 28 PM

#### To Test:
- yarn start:reactdev
- Navigate to Drawer->Drawer->API Docs (tab)

### CODE REVIEW

**Comment by huayunh** [comment]
I saw this off-white background color around examples that should not be here.

All "Examples" tabs shall be pure white in light theme (white50), and all "Playground" tabs shall be white200 in light theme.

image

**Comment by emclaug2** [comment]
I notice a difference between the md tables between angular and react.

React:
image


Angular: 
image

The biggest difference is the alternating row color & vertical borders.  Here's the styles I used to remove the columns from the table. 
```
table {
                margin-bottom: 8px;
                width: 100%;
                box-sizing: border-box;
                border-collapse: collapse;
                tr th {
                    border: unset;
                    border-left: 0;
                    border-right: 0;
                }
                tr {
                    &:last-of-type {
                        border-bottom: 0;
                    }
                }
}
```

File: docs/src/__configuration__/markdownMapping.tsx [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+                }}
+                {...props}
+            />
+        );
+    },
+    table: (props: TableProps): JSX.Element => {
+        return (
+            <Box
+                sx={{
+                    mx: '43px',
```
Comment (ektasawant) [inline docs/src/__configuration__/markdownMapping.tsx:233]: Will it work with multiple of 8 instead of 43px?

**Comment by emclaug2** [comment]
@bkarambe  feel free to tag me or @huayunh after these visual changes are implemented.

**Comment by huayunh** [comment]
Two more comments:

image

File: docs/src/__configuration__/markdownMapping.tsx [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+            <Typography
+                component={'blockquote'}
+                sx={{
+                    pr: 2,
+                    mb: 1,
+                    borderLeft: ' 4px solid #DDDDDD',
+                    p: '0 15px',
+                    m: '15px 0',
+                    color: '#666666',
+                    ...REGULAR_WIDTH_STYLE,
```
Comment (huayunh) [inline docs/src/__configuration__/markdownMapping.tsx:170]: These are a lot of hard-coded colors. Can you find the closest color in our theme colors and use those instead?

**Comment by huayunh** [comment]
I also noticed a lot of hard-coded styles in `docs/src/__configuration__/markdownMapping.tsx`. They are fine for now, but in the future when we define the MDX styles even more carefully and introduce dark themes, these are going to be changed.

**Comment by bkarambe** [comment]
@huayunh @emclaug2 @ektasawant Updated the PR.

**Comment by ektasawant** [comment]
Can we use `font-family: monospace` similar to angular?
Screenshot 2022-08-08 at 3 31 42 PM

**Comment by bkarambe** [comment]
> Can we use `font-family: monospace` similar to angular? Screenshot 2022-08-08 at 3 31 42 PM

Yes the font family already is **monospace** 
Screenshot 2022-08-08 at 4 00 15 PM

**Comment by ektasawant** [comment]
> @huayunh @emclaug2 @ektasawant Updated the PR.

@huayunh , @emclaug2 if everything is fine then can we merge it?

File: docs/src/__configuration__/markdownMapping.tsx [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+                    overflow: 'auto',
+                    boxSizing: 'border-box',
+                    mb: 2,
+
+                    table: {
+                        p: '6px 13px',
+                        textAlign: 'left',
+                        borderCollapse: 'collapse',
+                        minWidth: 750,
+                        border: `1px solid ${Colors.black[100]}`,
```
Comment (huayunh) [inline docs/src/__configuration__/markdownMapping.tsx:228]: Sorry for not noticing this earlier. The table border color should be the same as `theme.palette.divider` — `rgba(66, 78, 84, 12%)`. Everything else looks good to me visually.

**Comment by ektasawant** [comment]
All PR comments are resolved , merging this PR.

**Review by ektasawant — APPROVED** [review]

**Pull request merged**